processing/src/main/java/org/apache/druid/query/planning/ExecutionVertex.java

```diff
@@ -33,30 +33,61 @@
 import org.apache.druid.query.scan.ScanQuery;
 import org.apache.druid.query.spec.MultipleIntervalSegmentSpec;
 import org.apache.druid.query.spec.QuerySegmentSpec;
+import org.apache.druid.segment.SegmentReference;
+import org.apache.druid.segment.VirtualColumn;
 import org.apache.druid.segment.join.JoinPrefixUtils;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 
 /**
- * Represents the native engine's execution vertex.
+ * Represents the native engine's execution vertex - the execution unit it may execute in one go.
```

Comment:
we should define what "one go" means here. I think it means something like: one call of either `CachingClusteredClient` or `LocalQuerySegmentWalker`. (It definitely doesn't mean one run of `ClientQuerySegmentWalker`, because that could run _anything_ in one call.)